portindex2json/portindex2json.tcl

```diff
@@ -64,7 +64,22 @@ proc remove_extra_braces {text} {
     return [regsub -all {[\{\}]} $text ""]
 }
 
-set fd [open [lindex $argv 0] r]
+for {set i 0} {$i < $argc} {incr i} {
+    set arg [lindex $argv $i]
+    switch -regex -- $arg {
+        {^-.+} {
+            if {$arg eq "-info"} {
+                incr i
+                set argument [split [lindex $argv $i] =]
+                set json_info([lindex $argument 0]) [::json::write string [lindex $argument 1]]
+            }
+        }
+        {$.PortIndex|i} {
```

Comment:
This is a regression that really must be fixed. Running the following no longer works, for example:
```
/opt/local/libexec/macports/bin/tclsh8.5 portindex2json.tcl ../../macports-ports/PortIndex
```
Please also add `portindex2json -help` (accepting `-h`, `-help`, `--help`) printing the invocation. The invocation should also be printed (and exit with no error / 0) when the command doesn't provide any arguments.

Reply:
> This is a regression that really must be fixed. Running the following no longer works, for example:
> 
> ```
> /opt/local/libexec/macports/bin/tclsh8.5 portindex2json.tcl ../../macports-ports/PortIndex
> ```

I am now thinking of making the `default` case to set file path. This would also allow to have filenames other than `PortIndex`. Any views regarding this? Thanks
> 
> Please also add `portindex2json -help` (accepting `-h`, `-help`, `--help`) printing the invocation. The invocation should also be printed (and exit with no error / 0) when the command doesn't provide any arguments.

Reply:
It sounds ok to me to treat any argument that doesn't start with `-` as filename. Then check if that filename exists first, complain if it doesn't (with a more useful error message than simply getting stack trace from tcl), and make sure that exactly one such argument is provided. If there was no filename specified or if there were two arguments which didn't start with dash, also catch and report that in one way or another.

Reply:
Thank you. I have done some changes and tried to catch many errors. I have also added `-help` (accepting `-h`, `-help`, `--help`).